src/client/ua_client.c

```diff
@@ -415,11 +445,9 @@ receiveResponse(UA_Client *client, void *response, const UA_DataType *responseTy
 
     UA_StatusCode retval = UA_STATUSCODE_GOOD;
     UA_DateTime now = UA_DateTime_nowMonotonic();
+    UA_DateTime maxDate = UA_DateTime_nowMonotonic() + ((UA_DateTime)timeout * UA_DATETIME_MSEC);
     do {
-        UA_UInt32 timeout2 = (UA_UInt32)((maxDate - now) / UA_DATETIME_MSEC);
-        if(maxDate < now)
-            timeout2 = 0;
-        retval = UA_SecureChannel_receive(&client->channel, &rd, processServiceResponse, timeout2);
+        retval = UA_SecureChannel_receive(&client->channel, &rd, processServiceResponse, timeout);
```

Comment:
implement correct timeout modification - here it stays the same in every loop iteration